src/systems/ackermann_steering/AckermannSteering.cc

```diff
@@ -923,14 +955,17 @@ void AckermannSteeringPrivate::UpdateVelocity(
     return;
   }
 
-  double leftDelta = leftSteeringJointAngle - leftSteeringPos->Data()[0];
-  double rightDelta = rightSteeringJointAngle - rightSteeringPos->Data()[0];
+  double leftDelta = leftSteeringPos->Data()[0] - leftSteeringJointAngle;
+  double rightDelta = rightSteeringPos->Data()[0] - rightSteeringJointAngle;
 
-  // Simple proportional control with a gain of 1
-  // Adding programmable PID values might be a future feature.
-  // Works as is for tested cases
-  this->leftSteeringJointSpeed = leftDelta;
-  this->rightSteeringJointSpeed = rightDelta;
+  // Simple PID control with settable gains.
+  this->leftSteeringJointSpeed = this->steerPosPid.Update(
+    leftDelta, _info.dt);
+  this->steerPosPid.Reset();
```

Comment:
Is there a reason for resetting `steerPosPid` here? If I'm not mistaken, if we reset it, the `D` and `I` terms of the PID will not do anything.

Reply:
I was resetting the `steerPosPid` because I was using the same PID for both steering joints and it was causing random motions. I didn't realise it would render the `I` and `D` terms useless. 

After giving it more thought, I think this will be solved if I create 2 PID controllers in `Configure` - one for each steering joint.

Would like to hear your thoughts on this.

Reply:
I think we should get the fix in for the steering angle, but I don't think we should necessarily couple that with the PID changes. I think these should be separate PRs.

Reply:
Makes sense. So all the changes will have to be rolled back. The easiest way I can see is to reset this branch to commit id 633ce72171e27e83bf2e0292c9998e036d5da3fc and then resolve the bug. 

Should I go ahead with this or is there any alternate method that you have thought of?

Reply:
I guess I would just pull latest commit on branch and then make the fix ontop of it: https://github.com/gazebosim/gz-sim/commit/6cfcd07c346a72546963ed56ade48daa828c2072

Reply:
And then push your current branch to another branch like ackermann-pid so you could reference it later, and push the changes ontop the other to this current branch so you don't lose history in this PR.